### ISSUE TITLE
Site editor: show color warning if poor contrast colors are picked

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#29457]
## Description

If you select a color for the site title that results in poor contrast, no notice is given. This is feedback from the[ FSE Outreach Program's second call for testing](https://make.wordpress.org/test/2021/02/18/fse-program-testing-call-2-build-a-homepage-with-site-editing-blocks/). 

## Step-by-step reproduction instructions

Please list the steps needed to reproduce the bug. For example:
1. Use a block based theme and the latest version of Gutenberg to setup the FSE experience.
2. Go to "Site Editor (beta)"
3. Select the Site Title block and change the block settings so that the color contrast is poor for the text (or vice versa for the background).
4. See no notice. 

## Expected behaviour

I expect a notice about poor contrast similar to what happens when you do so in the post/page editor:

<img width="284" alt="Screen Shot 2021-03-01 at 3 53 24 PM" src="https://user-images.githubusercontent.com/26996883/109569816-483c4000-7aa6-11eb-93e0-e0bf64ed358b.png">

## Actual behaviour

No warning given. 

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/26996883/109569843-50947b00-7aa6-11eb-8847-9d65a160b5ec.mov

## WordPress information
- WordPress version: 5.6.2
- Gutenberg version: 10.0.2
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? TT1 Blocks

**Comment by Quintis1212** [comment]
Hello ) I think I can fix this )

**Comment by Quintis1212** [comment]
I see that notice about poor contrast and color options not connected , and they are called from react-dom :
![Знімок екрана з 2021-03-02 11-09-08](https://user-images.githubusercontent.com/34488718/109625364-edf3bb80-7b47-11eb-92fb-d57a161089d4.png)
Who can help to fix this ?What i must change in the code ?

**Comment by annezazu** [comment]
@nosolosw perhaps you can share some insight? If not, no worries :)

**Comment by nosolosw** [comment]
Thanks for looking into this @Quintis1212 !

I took a quick look at this and I was able to reproduce with any block that uses the color panel we provide (for example, paragraph). Apparently, the post editor and the site editor use the same mechanism ([this](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/style.js#L185)), so I'm not sure why the color contrast notice doesn't work for the site editor off the top of my head. Would that link be helpful to start to tracking things down, @Quintis1212?

**Comment by Quintis1212** [comment]
Thank you for help  @nosolosw ! I wil continue to fix this bug with help of  your info )

**Comment by Quintis1212** [comment]
@nosolosw   I finded out that there is a function which works incorrect :  
path : gutenberg/packages/block-editor/src/utils/dom.js
`
export function getBlockDOMNode( clientId, doc ) {
	console.log(clientId, doc )
	console.log(document.getElementById( 'block-' + clientId ))
	console.log(doc.getElementById( 'block-' + clientId ))
	return doc.getElementById( 'block-' + clientId );
}

`

This function can not find element in site editor : 
![Знімок екрана з 2021-03-03 16-00-03](https://user-images.githubusercontent.com/34488718/109816852-e4e31700-7c39-11eb-9249-cb0b8dd8708a.png)

**Comment by nosolosw** [comment]
Ah, I see. The color contrast feature uses [the DOM to find the block node](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/color-panel.js#L32). However, this is problematic here because the editor canvas and the sidebar live in different documents (the editor is iframed) ― that's why the colorsDetectionElement is never found here but it is in the post editor. Does this provide any help to continue investigating the fix? I'm going to pull @ellatrix thoughts on this as well.

**Comment by nosolosw** [comment]
I've taken the liberty of exchanging "Site Title" for "Site Editor" from the issue title and the label, as I think it reflects the issue is wider and affects all blocks.

**Comment by aristath** [comment]
I had another ticket for this on https://github.com/WordPress/gutenberg/issues/28782 (for global styles), closed as a duplicate.

**Comment by Quintis1212** [comment]
It seems we need to use refs for detecting elements colors , I can not save document object in local storage 
 path - gutenberg/packages/edit-site/src/components/block-editor/index.js
`
	const thisDOM = window.document;
	localStorage.setItem('editor-dom', thisDOM);
`

 path : gutenberg/packages/block-editor/src/utils/dom.js
`
export function getBlockDOMNode( clientId, doc ) {

	//setTimeout( ()=>{
		//console.trace(1)
		//console.log(clientId, doc )
		//console.log('block-' + clientId )
		//console.log(window.document)
		//console.log(document.getElementById( 'block-' + clientId ))
		//console.log(editorDOM().getElementById( 'block-' + clientId ))
		const editorDOM = localStorage.getItem('editor-dom');
	24	console.log(editorDOM+"editor dom")
	25	console.log( document.getElementById( 'block-' + clientId ).bind(editorDOM)  )
		return doc.getElementById( 'block-' + clientId );
	//}, 1000)
}
`

![Знімок екрана з 2021-03-04 13-36-13](https://user-images.githubusercontent.com/34488718/109958290-bd9a5180-7cee-11eb-8ab1-54bb130d6e35.png)

**Comment by Quintis1212** [comment]
@annezazu , @nosolosw , @aristath  please , can any one help me for rewiev for this issue -  https://github.com/WordPress/gutenberg/pull/29534 ,  there is no set uped reviewers by default ) This is just one line of code for enhancement accessibility )

**Comment by ItsJonQ** [comment]
> However, this is problematic here because the editor canvas and the sidebar live in different documents (the editor is iframed) 

@nosolosw I remember seeing some refactors to swap `document` to something like `node.ownerDocument.defaultView`.
However, this specific `getBlockDOMNode()` call doesn't receive a `node` to check against 🤔 .

**Comment by ellatrix** [comment]
Looking into this
